ts-types/reg-exp/ensure.d.ts

```diff
@@ -0,0 +1,4 @@
+import { EnsureOptions } from '../ensure';
+
+declare function ensureRegExp(value: any, options?: EnsureOptions): RegExp;
```

Comment:
I believe here we also can do `enureRegExp<T>..` as in all cases input object will be returned

Reply:
I'm confuse on this one. Can `ensureRegExp` return anything else that is not a `RegExp`?

Reply:
Technically not, but it can return `RegExp` from a different realm, I'm not sure if in TS, objects that come from two different native `RegExp` classes will be accepted as `RegExp` type, will they?

Reply:
If you mean cases like:
- class MultiRegExp2 extends RegExp
- /foo/

yes, they are accepted as `RegExp`. 
In case you meant something else pls put some test case.

Reply:
I mean't different instances of native `RegExp` class, which can come from other realms, e.g. in browsers you can create other realms by creating iframe:

```javascript
const otherRealm = document.body.appendChild(document.createElement('iframe')).contentWindow;
console.log(otherRealm.RegExp); // Native RegExp class
console.log(RegExp === otherRealm.RegExp); // false
```

Question is, wether TS will recognize a `new otherRealm.RegExp()` as value of `RegExp` type

Reply:
If that's not the case, it'll be safer to use `enureRegExp<T>` notation I think

Reply:
Got it. Let me have some test.

Reply:
@medikoo, apparently when you get `contentWindow` the type is `Window | null` which does not have `RegExp` in it.
Thus for type purpose you have to [type assert](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions) to `Window & typeof globalThis`, which is the same type of `window`.

The result is that, from a type point of view,  `otherRealm.RegExp` has the same type of `window.RegExp`.

```ts
const iframe:HTMLIFrameElement = document.body.appendChild(document.createElement('iframe'));
var otherRealm = (iframe.contentWindow as Window & typeof globalThis); // Cast to Window & typeof globalThis else it would be Window | null and RegExp doesn't exist in Window

function test(a: RegExp): RegExp {
   return a;
};
test(otherRealm?.RegExp('asdf'));

```

Reply:
Great thanks @borracciaBlu for clarifying on that